src/jobs/run_codacy_plugins_test.yml

```diff
@@ -0,0 +1,36 @@
+description: "Runs codacy-plugins-test on a tool"
+
+executor: machine
+
+parameters:
+  codacy_plugins_test_version:
+    type: string
+    description: "codacy-plugins-test version"
+
+environment:
+  CODACY_PLUGINS_TEST_VERSION: << parameters.codacy_plugins_test_version >>
+
+steps:
+  - attach_workspace:
+      at: ~/
+  - restore_cache:
+      key: codacy-plugins-test-cache-$CODACY_PLUGINS_TEST_VERSION
+  - attach_workspace:
+      at: ~/
+  - run:
+      command: |
+        mkdir -p codacy-plugins-test
+        cd codacy-plugins-test
+        FILENAME=codacy-plugins-test-$CODACY_PLUGINS_TEST_VERSION
+        LINK="https://bintray.com/codacy/Binaries/download_file?file_path=$CODACY_PLUGINS_TEST_VERSION%2Fcodacy-plugins-test-linux"
+        if [ ! -f "$FILENAME" ]; then
+          wget $LINK -O $FILENAME
+          chmod +x $FILENAME
+        fi
+        docker load --input ~/workdir/docker-image.tar
+        ./$FILENAME json $CIRCLE_PROJECT_REPONAME:latest
```

Comment:
Its a good idea to make a parameter from `CIRCLE_PROJECT_REPONAME` environment variable and left that value as a default value. When I done a PR to the orbs about the plugins test, I made some additional parameters that could be good, like:

```yml
  run_json_test:
    type: boolean
    default: true
    description: Run 'json' test in plugins test
  run_pattern_test:
    type: boolean
    default: true
    description: Run 'pattern' test in plugins test
  run_plugin_test:
    type: boolean
    default: true
    description: Run 'plugin' test in plugins test
```

See [this][1] .

[1]: https://github.com/codacy/codacy-orbs/pull/7/files#diff-ec46400109ec431f35e77839e1865c00R5

Reply:
We test every tool the same way. I prefer no other parameters and run directly.

Reply:
For example, we use it on `codacy-scalameta`, because we have tests for every pattern.

Reply:
Done 👍

Reply:
If you see [here](https://github.com/codacy/codacy-orbs/pull/7/files#diff-ec46400109ec431f35e77839e1865c00R74), circle ci already have a `when` keyword for conditional parameters. With this, you have clean seperate stages on the UI instead of bash if statement.